lib/interactor/context.rb

```diff
@@ -121,10 +121,22 @@ def failure?
     #
     # Raises Interactor::Failure initialized with the Interactor::Context.
     def fail!(context = {})
-      error_context = Context.new
-      context.each { |key, value| error_context[key.to_sym] = value }
-      error_context.failure = @failure = true
-      raise Failure, error_context
+      context.each { |key, value| self[key.to_sym] = value }
+
+      redaction_message = 'REDACTED'
+      self.each_pair do |name, value|
```

Comment:
I'd use `|key, value|`, it seems to be a more "common" ruby convention (although there's no real standard, to be honest)